src/viam/sdk/common/utils.cpp

```diff
@@ -82,5 +85,14 @@ google::protobuf::Duration to_proto(const std::chrono::microseconds& duration) {
     return proto;
 }
 
+void set_logger_severity_from_args(int argc, char** argv) {
+    if (argc >= 3 && strcmp(argv[2], "--log-level=debug") == 0) {
```

Comment:
This seems pretty brittle. Specifically, the requirement that `log-level` _must_ be the third arg. Could we instead filter through with [find](https://en.cppreference.com/w/cpp/algorithm/find)?

I could be convinced to leave it as is if you feel strongly about maintaining parity with `Go` on this.

Reply:
> Could we instead filter through with [find](https://en.cppreference.com/w/cpp/algorithm/find)?

Certainly could! 

Python and Go do have equally brittle functions. My thought here is that at some point we'll add more exhaustive logging with many levels in the RDK and SDKs. This function will then probably change to parse for `--log-level` in any arg position, and we'll have more complex switch logic to map the string argument to a logging level. I was thinking we would make the function less brittle at that point in time.

Reply:
Got it. Happy to leave this for a future date if you'd rather make all those improvements at once :)

Reply:
Sounds good 😃 would prefer to leave this as-is then for now.